dts/arm/atmel/samd21.dtsi

```diff
@@ -7,6 +7,22 @@
 #include <atmel/samd.dtsi>
 
 / {
+	clocks {
+		fdpll: fdpll {
```

Comment:
samd5x has two of those - would it make sense to call it `fdpll0` to make sharing code easier?
I'm currently writing a separate `soc_samd5x.c` but I want to see if if I can merge it with `soc_samd2x.c` later.

Reply:
I am slightly inclined to keep them separate so you can do constructs like:

```C
#ifdef DT_FIXED_CLOCK_FDPLL_CLOCK_FREQUENCY
/* SAMD2x ASF referencing code */
#endif
#ifdef DT_FIXED_CLOCK_FDPLL0_CLOCK_FREQUENCY
/* SAMD5x ASF referencing code */
#endif
#ifdef DT_FIXED_CLOCK_FDPLL1_CLOCK_FREQUENCY
/* SAMD5x ASF referencing code */
#endif
```

The counter argument is that the SAMD5x ASF headers do define some aliases (e.g. `GCLK_SOURCE_FDPLL` for `GCLK_SOURCE_FDPLL0`) so some of that is unneeded (bad example, since you'd still need that one; the naming is slightly different).  

Basically in nearly all cases where this is referenced, there's going to be ASF defines, and I haven't really tried to port the SAMD5x code to figure out how much of those can be shared without an ifdef soup.

So, if you expect that the sharing would be simpler with the ifdefs in the inner blocks, I can certainly change it.

Reply:
I have a [very much work in progress](https://github.com/benpicco/zephyr/tree/same54-dts-clock) port to SAM D5x.
The structure is mostly the same, but register names change all over the place, so it's probably cleaner to keep them separate for now.